lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -98,7 +122,10 @@ public ParquetReader(
             ParquetDataSource dataSource,
             DateTimeZone timeZone,
             AggregatedMemoryContext systemMemoryContext,
-            ParquetReaderOptions options)
+            ParquetReaderOptions options,
+            Predicate parquetPredicate,
+            List<ColumnIndexStore> blockIndexStores,
```

Comment:
To fit with the conventions of Trino, these nullable parameters should almost certainly be `Optional` instead.